airflow/api_fastapi/core_api/routes/ui/grid.py

```diff
@@ -0,0 +1,229 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import collections
+import itertools
+from typing import Annotated
+
+from fastapi import Depends, HTTPException, Request, status
+from sqlalchemy import select
+
+from airflow import DAG
+from airflow.api_fastapi.common.db.common import SessionDep, paginated_select
+from airflow.api_fastapi.common.parameters import (
+    OptionalDateTimeQuery,
+    QueryDagRunRunTypesFilter,
+    QueryDagRunStateFilter,
+    QueryIncludeDownstream,
+    QueryIncludeUpstream,
+    QueryLimit,
+    QueryOffset,
+    Range,
+    RangeFilter,
+    SortParam,
+)
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.datamodels.ui.grid import (
+    GridDAGRunwithTIs,
+    GridResponse,
+)
+from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.api_fastapi.core_api.services.ui.grid import (
+    fill_task_instance_summaries,
+    get_child_task_map,
+    get_dag_run_sort_param,
+    get_task_group_map,
+)
+from airflow.models import DagRun, TaskInstance
+from airflow.models.dagrun import DagRunNote
+from airflow.models.taskinstance import TaskInstanceNote
+
+grid_router = AirflowRouter(prefix="/grid", tags=["Grid"])
+
+
+@grid_router.get(
+    "/{dag_id}",
+    include_in_schema=False,
+    responses=create_openapi_http_exception_doc([status.HTTP_400_BAD_REQUEST, status.HTTP_404_NOT_FOUND]),
+)
+def grid_data(
+    dag_id: str,
+    run_types: QueryDagRunRunTypesFilter,
+    run_states: QueryDagRunStateFilter,
+    session: SessionDep,
+    offset: QueryOffset,
+    request: Request,
+    limit: QueryLimit,
+    order_by: Annotated[
+        SortParam,
+        Depends(
+            SortParam(
+                ["logical_date", "data_interval_start", "data_interval_end", "start_date", "end_date"], DagRun
+            ).dynamic_depends()
+        ),
+    ],
+    include_upstream: QueryIncludeUpstream = False,
+    include_downstream: QueryIncludeDownstream = False,
+    logical_date_gte: OptionalDateTimeQuery = None,
+    logical_date_lte: OptionalDateTimeQuery = None,
+    root: str | None = None,
+) -> GridResponse:
+    """Return grid data."""
+    dag: DAG = request.app.state.dag_bag.get_dag(dag_id)
+    if not dag:
+        raise HTTPException(status.HTTP_404_NOT_FOUND, f"Dag with id {dag_id} was not found")
+
+    date_filter = RangeFilter(
+        Range(lower_bound=logical_date_gte, upper_bound=logical_date_lte),
+        attribute=DagRun.logical_date,
+    )
+    # Retrieve, sort and encode the previous DAG Runs
```

Comment:
I don't think encoding is done here.

```suggestion
    # Retrieve, sort the previous DAG Runs
```

Reply:
Thanks, removed!